saleor/warehouse/models.py

```diff
@@ -182,16 +182,7 @@ def delete(self, *args, **kwargs):
 
 class StockQuerySet(models.QuerySet):
     def annotate_available_quantity(self):
-        return self.annotate(
-            available_quantity=F("quantity")
-            - Coalesce(
-                Sum(
-                    "allocations__quantity_allocated",
-                    filter=Q(allocations__quantity_allocated__gt=0),
-                ),
-                0,
-            )
-        )
+        return self.annotate(available_quantity=F("quantity") - F("quantity_allocated"))
```

Comment:
Maybe we should drop this function and consider every usage of them because in the new solution we are able to write some queries without annotating it.

Reply:
Good idea, I've dropped it entirely.